core/src/value/partial.rs

```diff
@@ -0,0 +1,873 @@
+            range,
+        }
+        .fail()
+    }
+}
+
+impl DicomDate {
+    /**
+     * Constructs a new `DicomDate` with year precision
+     * (YYYY)
+     */
+    pub fn from_y(year: u16) -> Result<DicomDate> {
+        check_component(DateComponent::Year, &year)?;
+        Ok(DicomDate::Year(year))
+    }
+    /**
+     * Constructs a new `DicomDate` with year and month precision
+     * (YYYYMM)
+     */
+    pub fn from_ym(year: u16, month: u8) -> Result<DicomDate> {
+        check_component(DateComponent::Year, &year)?;
+        check_component(DateComponent::Month, &month)?;
+        Ok(DicomDate::Month(year, month))
+    }
+    /**
+     * Constructs a new `DicomDate` with a year, month and day precision
+     * (YYYYMMDD)
+     */
+    pub fn from_ymd(year: u16, month: u8, day: u8) -> Result<DicomDate> {
+        check_component(DateComponent::Year, &year)?;
+        check_component(DateComponent::Month, &month)?;
+        check_component(DateComponent::Day, &day)?;
+        Ok(DicomDate::Day(year, month, day))
+    }
+}
+
+impl TryFrom<&NaiveDate> for DicomDate {
+    type Error = Error;
+    fn try_from(date: &NaiveDate) -> Result<Self> {
+        let year: u16 = date.year().try_into().context(Conversion)?;
+        let month: u8 = date.month().try_into().context(Conversion)?;
+        let day: u8 = date.day().try_into().context(Conversion)?;
+        Ok(DicomDate::from_ymd(year, month, day)?)
+    }
+}
+
+impl fmt::Display for DicomDate {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            DicomDate::Year(y) => write!(f, "{:04}-MM-DD", y),
+            DicomDate::Month(y, m) => write!(f, "{:04}-{:02}-DD", y, m),
+            DicomDate::Day(y, m, d) => write!(f, "{:04}-{:02}-{:02}", y, m, d),
+        }
+    }
+}
+
+impl DicomTime {
+    /**
+     * Constructs a new `DicomTime` with hour precision
+     * (HH).
+     */
+    pub fn from_h(hour: u8) -> Result<DicomTime> {
+        check_component(DateComponent::Hour, &hour)?;
+        Ok(DicomTime::Hour(hour))
+    }
+
+    /**
+     * Constructs a new `DicomTime` with hour and minute precision
+     * (HHMM).
+     */
+    pub fn from_hm(hour: u8, minute: u8) -> Result<DicomTime> {
+        check_component(DateComponent::Hour, &hour)?;
+        check_component(DateComponent::Minute, &minute)?;
+        Ok(DicomTime::Minute(hour, minute))
+    }
+
+    /**
+     * Constructs a new `DicomTime` with hour, minute and second precision
+     * (HHMMSS).
+     */
+    pub fn from_hms(hour: u8, minute: u8, second: u8) -> Result<DicomTime> {
+        check_component(DateComponent::Hour, &hour)?;
+        check_component(DateComponent::Minute, &minute)?;
+        check_component(DateComponent::Second, &second)?;
+        Ok(DicomTime::Second(hour, minute, second))
+    }
+
+    /**
+     * Constructs a new `DicomTime` with hour, minute, second and second fraction
+     * precision (HHMMSS.FFFFFF).
+     */
+    pub fn from_hmsf(
```

Comment:
This one feels a bit hard to use, since the given `frac_precision` could be inconsistent with `fraction`. I would suggest separating this it to one that takes milliseconds and another one that takes microseconds. If this function is to stay, it should be carefully documented to better explain how to use `frac_precision`.